### PR TITLE
[WIP] Offset external/v2

### PR DESCRIPTION
**Pull request by cryogenian**
Part of https://github.com/precog/quasar/pull/4734 meta-pr 

+ Removed parsing format config field
+ Added one more parsing commitable record option"
+ Introduced seek evaluator (which actually `seek`s using persisted offsets) 

Still WIP as quasar|mimir could be changed, any thoughts and suggestions are very appreciated

### CODE REVIEW

File: core/src/main/scala/quasar/datasource/kafka/Config.scala [hunk 1]
```diff
@@ -22,15 +22,13 @@ import argonaut.Argonaut._
...
 import cats.syntax.either._
-import quasar.connector.DataFormat
 
 case class Config(
   bootstrapServers: NonEmptyList[String],
   groupId: String,
   topics: NonEmptyList[String],
   tunnelConfig: Option[TunnelConfig],
   decoder: Decoder,
-  format: DataFormat
```
[the remark this answers is not in the document]
Comment (cryogenian) [reply core/src/main/scala/quasar/datasource/kafka/Config.scala:33]: All data in kafka-ds we could parse is actually ldjson. It's hard to imagine anyone using csv here. Also @alissapajer could verify

File: core/src/main/scala/quasar/datasource/kafka/KafkaConsumerBuilder.scala [hunk 2]
```diff
@@ -84,6 +86,24 @@ object KafkaConsumerBuilder {
...
 
+  def AsJson[F[_]]: RecordDecoder[F, Array[Byte], Array[Byte]] =
+    (record: CommittableConsumerRecord[F, Array[Byte], Array[Byte]]) => {
+      // here we suppose that both `value` and `key` bytes are valid jsons
+      val value = Option(record.record.value).getOrElse("null".getBytes)
+      val key = Option(record.record.key).getOrElse("null".getBytes)
+      val prefixBytes = "{\"key\":".getBytes
+      val midBytes = ",\"value\":".getBytes
+      val suffixBytes = "}".getBytes
+      val arr = ArrayBuilder.make[Byte]
```
[the remark this answers is not in the document]
Comment (cryogenian) [reply core/src/main/scala/quasar/datasource/kafka/KafkaConsumerBuilder.scala:97]: Thanks!

File: core/src/main/scala/quasar/datasource/kafka/KafkaConsumerBuilder.scala [hunk 3]
```diff
@@ -84,6 +86,24 @@ object KafkaConsumerBuilder {
     (record: CommittableConsumerRecord[F, Array[Byte], Array[Byte]]) =>
       Stream.chunk(Chunk.bytes(Option(record.record.value).getOrElse(Array.empty)))
 
+  def AsJson[F[_]]: RecordDecoder[F, Array[Byte], Array[Byte]] =
```
Comment (alissapajer) [inline core/src/main/scala/quasar/datasource/kafka/KafkaConsumerBuilder.scala:89]: Why do we need to add a new decoder type `AsJson` that includes both the key and value?

Comment (cryogenian) [reply core/src/main/scala/quasar/datasource/kafka/KafkaConsumerBuilder.scala:89]: I can remove it for sure, but at least for me it seems pretty weird that we can't have both keys and values inside data analysis. What's the point of having only keys or only values? If I send something like `key: 2021-01-26:22:34  value: { message: "trololo"}` and then `key: 2021-01-26:22:36 value: {message: "trololo"}` the information about this message was sent twice is erased, it's just `Ok, two`. I can remove it if it's not needed though.

Comment (alissapajer) [reply core/src/main/scala/quasar/datasource/kafka/KafkaConsumerBuilder.scala:89]: That makes sense. Let's keep this here for now. I'll try to find out exactly what the JSON on the Kafka topic will look like for this initial customer use case.

Comment (alissapajer) [reply core/src/main/scala/quasar/datasource/kafka/KafkaConsumerBuilder.scala:89]: When we initially implemented the Kafka datasource, we didn't have a concrete use case in mind, and so we decided to support both JSON and CSV, at the cost of only decoding the key or the value, exclusive. Now that we have a use case, we can let it shape the decision we make here.

File: core/src/test/scala/quasar/datasource/kafka/SeekConsumerSpec.scala [hunk 4]
```diff
@@ -14,6 +14,22 @@
  * limitations under the License.
  */
 
+/*
```
Comment (wemrysi) [inline core/src/test/scala/quasar/datasource/kafka/SeekConsumerSpec.scala:17]: Duplicate header.

**Comment by cryogenian** [comment]
+ Removed duplicated header

**Pull request closed without merging**